pkg/controllers/addons_controller.go

```diff
@@ -103,13 +90,24 @@ func (r *AddonsLayerReconciler) processAddonLayer(l layers.Layer) error { // nol
 		l.SetDelayedRequeue()
 		return nil
 	}
+	return nil
+}
+
+func (r *AddonsLayerReconciler) processApply(l layers.Layer) error {
+	if l.IsUpdated() {
+		return nil
+	}
+
+	ctx := r.Context
+	applier := r.Applier
 
 	applyIsRequired, err := applier.ApplyIsRequired(ctx, l)
 	if err != nil {
 		// TODO - we might want to add some sort of error handling here
 		return err
 	} else if applyIsRequired {
 		if !l.DependenciesDeployed() {
+			l.SetStatusPending()
```

Comment:
This is not needed, DependenciesDeployed() sets the status to ApplyPending if one of the layer's dependsOn items are not deployed. Doing it inside DependenciesDeployed() allows us to capture the reason for the the ApplyPending status, i.e. it sets the reason field to say what it is waiting for and why.

Reply:
👍

Reply:
Mainly what I'd wanted to accomplish here was to set `l.updated = true`, which I can now see is triggered by the status updates in the `isOtherDeployed` function in `layers.go` lines 229 and 235.

I'm fine with doing it that way, but I'd consider the status update in a checking function something of an unexpected side-effect to the stated purpose of the function.  It's a little confusing, as evidenced by my impulse to add `SetStatusPending` here.

I'll undo this change regardless - we can always debate over functional programming paradigms over a drink sometime